core/string/ustring.cpp

```diff
@@ -1644,6 +1645,32 @@ String String::hex_encode_buffer(const uint8_t *p_buffer, int p_len) {
 	return ret;
 }
 
+Vector<uint8_t> String::hex_decode() const {
+	const String *s = this;
+	Vector<uint8_t> out;
+	ERR_FAIL_COND_V_MSG(s->size() % 2 == 0, out, "Hexadecimal string of uneven length.");
```

Comment:
```suggestion
	ERR_FAIL_COND_V_MSG(s->size() % 2 != 0, out, "Hexadecimal string of uneven length.");
```
Error macros work in the opposite way to `assert()`.

Reply:
no, it works like this. i think its because of the null byte or something.

Reply:
You should use `length()`:

https://github.com/godotengine/godot/blob/0bd143dff90e970e6a73cc53d331312fcf48acbc/core/string/ustring.h#L119